pkg/deployer/lib/controller.go

```diff
@@ -197,6 +200,12 @@ func (c *controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return returnAndLogReconcileResult(logger, *hookRes), nil
 	}
 
+	lsCtx := &lsv1alpha1.Context{}
+	// todo: check for real repository context. Maybe overwritten by installation.
```

Comment:
Reading this I have a feeling this could be something that leads to issues on user side by picking the wrong context.
What is currently stopping us from finding the "real" repository context? Is it a matter of missing APIs?

Reply:
> What is currently stopping us from finding the "real" repository context? Is it a matter of missing APIs?

The initial idea was that there is default context defined that is used by all installation.
Only for dev or debugging purposes the repository context directly in the installation should be set.
The same idea is then also used in the implemenation with the context object.
The context object is there to namespaced default context and in a production environment the repository should never be set in the installation but always be read from the context.

Coming now to deploy items, which are reconciled by deployers, there is now the problem that if a context is overwritten by the installation this is not reflected int he context object.
And the issue is that the deployer should not know about an installation. The only contract should be a deploy/target and context.